_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -41,14 +42,38 @@ static bool IsSameVideoParam(mfxVideoParam *newPar, mfxVideoParam *oldPar);
 inline
 bool CheckHardwareSupport(VideoCORE *p_core, mfxVideoParam *p_video_param)
 {
-    (void)p_core;
+    if(!p_core) return false;
+    if(!p_video_param) return false;
 
-#ifdef ANDROID
-    if (p_video_param->mfx.FrameInfo.Width > 4096 || p_video_param->mfx.FrameInfo.Height > 4096)
+    GUID guid;
+
+    switch(p_video_param->mfx.CodecProfile)
     {
-        return false;
+    case(MFX_PROFILE_VP9_0):
+    {
+        guid = DXVA_Intel_ModeVP9_Profile0_VLD;
+        break;
     }
-#endif
+    case(MFX_PROFILE_VP9_1):
+    {
+        guid = DXVA_Intel_ModeVP9_Profile1_YUV444_VLD;
+        break;
+    }
+    case(MFX_PROFILE_VP9_2):
+    {
+        guid = DXVA_Intel_ModeVP9_Profile2_10bit_VLD;
+        break;
+    }
+    case(MFX_PROFILE_VP9_3):
+    {
+        guid = DXVA_Intel_ModeVP9_Profile3_YUV444_10bit_VLD;
+        break;
+    }
+    default: return false;
+    }
+
+    mfxStatus mfxSts = p_core->IsGuidSupported(guid, p_video_param);
+    MFX_CHECK_STS(mfxSts);
```

Comment:
this, function return bool, but this macro return mfxStatus. Replace it with another one
MFX_CHECK(mfxSts == MFX_ERR_NONE, false);